src/org/zaproxy/zap/extension/spiderAjax/AjaxSpiderDialog.java

```diff
@@ -217,13 +213,49 @@ public void save() {
     	this.extension.spiderSite(this.startNode, this.getBoolValue(FIELD_IN_SCOPE), params);
     }
 
+    /**
+     * Gets the selected browser.
+     *
+     * @return the selected browser, {@code null} if none selected
+     */
+    private Browser getSelectedBrowser() {
+        if (isEmptyField(FIELD_BROWSER)) {
+            return null;
+        }
+
+        String browserName = this.getStringValue(FIELD_BROWSER);
+        List<BrowserUI> browserList = getExtSelenium().getBrowserUIList();
+        for (BrowserUI bui : browserList) {
+            if (browserName.equals(bui.getName())) {
+                return bui.getBrowser();
+            }
+        }
+        return null;
+    }
+
     @Override
     public String validateFields() {
 
         if (this.startNode == null) {
             return Constant.messages.getString("spiderajax.scandialog.nostart.error");
         }
 
+        Browser selectedBrowser = getSelectedBrowser();
+        if (selectedBrowser == null) {
+            return null;
+        }
+
+        if (Browser.PHANTOM_JS.getId() == selectedBrowser.getId()) {
+            try {
+                String host = startNode.getHistoryReference().getURI().getHost();
+                if ("localhost".equalsIgnoreCase(host) || "127.0.0.1".equals(host) || "[::1]".equals(host)) {
```

Comment:
Based on the docs [1] and (some) tests that's the only affected address.

[1] http://doc.qt.io/qt-5.6/qhostaddress.html#SpecialAddress-enum